source/server/config/http/router.cc

```diff
@@ -12,23 +12,42 @@ namespace Envoy {
 namespace Server {
 namespace Configuration {
 
-HttpFilterFactoryCb RouterFilterConfig::createFilterFactory(const Json::Object& json_config,
-                                                            const std::string& stat_prefix,
-                                                            FactoryContext& context) {
-  json_config.validateSchema(Json::Schema::ROUTER_HTTP_FILTER_SCHEMA);
-
+HttpFilterFactoryCb RouterFilterConfig::createRouterFilter(bool dynamic_stats,
+                                                           bool start_child_span,
+                                                           const std::string& stat_prefix,
+                                                           FactoryContext& context) {
   Router::FilterConfigSharedPtr config(new Router::FilterConfig(
       stat_prefix, context.localInfo(), context.scope(), context.clusterManager(),
       context.runtime(), context.random(),
       Router::ShadowWriterPtr{new Router::ShadowWriterImpl(context.clusterManager())},
-      json_config.getBoolean("dynamic_stats", true),
-      json_config.getBoolean("start_child_span", false)));
+      dynamic_stats, start_child_span));
 
   return [config](Http::FilterChainFactoryCallbacks& callbacks) -> void {
     callbacks.addStreamDecoderFilter(std::make_shared<Router::ProdFilter>(*config));
   };
 }
 
+HttpFilterFactoryCb RouterFilterConfig::createFilterFactory(const Json::Object& json_config,
+                                                            const std::string& stat_prefix,
+                                                            FactoryContext& context) {
+  json_config.validateSchema(Json::Schema::ROUTER_HTTP_FILTER_SCHEMA);
+  return createRouterFilter(json_config.getBoolean("dynamic_stats", true),
```

Comment:
Per @zuercher can we convert to proto and pass? This will be easier in the future as we add more config.

Reply:
You can look at my change here: https://github.com/envoyproxy/envoy/compare/master...turbinelabs:implement-upstream-log#diff-ac294912df489e3b4f8094bdb1e91918

I can also convert this when I merge, so I added a translate function in FilterJson, a schema in config_schema.cc, tests, etc... It's a whole thing.

Reply:
@zuercher maybe you and @rshriram can coordinate over slack briefly? Seem like he could just copy part of your PR or could split off part of your PR. Either way.